pyscf/pbc/df/rsdf_helper.py

```diff
@@ -0,0 +1,1297 @@
+# +++++++ Use the default for the following unless you know what you are doing
+              lmp=True, lasympt=True,
+              eta_correct=True, R_correct=False, vol_correct=False,
+# -------
+# +++++++ debug options
+              no_screening=False,   # set Rcuts to effectively infinity
+# -------
+            ):
+    log = logger.Logger(cell.stdout, cell.verbose)
+
+    t1 = np.asarray([logger.process_clock(), logger.perf_counter()])
+
+    intor = "int2c2e"
+    intor, comp = mol_gto.moleintor._get_intor_and_comp(
+                                            cell._add_suffix(intor), None)
+    assert(comp == 1)
+
+# prescreening data
+    if precision is None: precision = cell.precision
+
+    refuniqshl_map, uniq_atms, uniq_bas, uniq_bas_loc = get_refuniq_map(cell)
+    Rcuts = get_2c2e_Rcut(uniq_bas, cell.vol, omega, precision,
+                          lmp=lmp, lasympt=lasympt,
+                          eta_correct=eta_correct, R_correct=R_correct,
+                          vol_correct=vol_correct)
+    Rcut2s = np.ones(Rcuts)*1e20 if no_screening else Rcuts**2.
+    atom_Rcuts = get_atom_Rcuts_2c(Rcuts, uniq_bas_loc)
+    cell_rcut = atom_Rcuts.max()
+    Ls = get_Lsmin(cell, atom_Rcuts, uniq_atms)
+    log.debug1("j2c prescreening: cell rcut %.2f Bohr  keep %d imgs",
+               cell_rcut, Ls.shape[0])
+    t1 = log.timer_debug1('prescrn warmup', *t1)
+# end prescreening data
+
+    if kpts is None:
+        kpts_lst = np.zeros((1,3))
+    else:
+        kpts_lst = np.reshape(kpts, (-1,3))
+    nkpts = len(kpts_lst)
+
+    if hermi == 0:
+        aosym = 's1'
+    else:
+        aosym = 's2'
+    fill = getattr(libpbc, 'PBCsr2c_fill_k'+aosym)
+    fintor = getattr(mol_gto.moleintor.libcgto, intor)
+    cintopt = lib.c_null_ptr()
+
+    pcell = copy.copy(cell)
+    pcell.precision = min(cell.precision, cell.precision)
+    pcell._atm, pcell._bas, pcell._env = \
+            atm, bas, env = mol_gto.conc_env(cell._atm, cell._bas, cell._env,
+                                             cell._atm, cell._bas, cell._env)
+    env[mol_gto.PTR_RANGE_OMEGA] = -abs(omega)
+    if shls_slice is None:
+        shls_slice = (0, cell.nbas, 0, cell.nbas)
+    i0, i1, j0, j1 = shls_slice[:4]
+    j0 += cell.nbas
+    j1 += cell.nbas
+    ao_loc = mol_gto.moleintor.make_loc(bas, intor)
+    ni = ao_loc[i1] - ao_loc[i0]
+    nj = ao_loc[j1] - ao_loc[j0]
+
+    out = np.empty((nkpts,comp,ni,nj), dtype=np.complex128)
+
+    expkL = np.asarray(np.exp(1j*np.dot(kpts_lst, Ls.T)), order='C')
+    drv = libpbc.PBCsr2c_k_drv
+
+    drv(fintor, fill, out.ctypes.data_as(ctypes.c_void_p),
+        ctypes.c_int(nkpts), ctypes.c_int(comp), ctypes.c_int(len(Ls)),
+        Ls.ctypes.data_as(ctypes.c_void_p),
+        expkL.ctypes.data_as(ctypes.c_void_p),
+        (ctypes.c_int*4)(i0, i1, j0, j1),
+        ao_loc.ctypes.data_as(ctypes.c_void_p), cintopt,
+        refuniqshl_map.ctypes.data_as(ctypes.c_void_p),
```

Comment:
ensure dtype of refuniqshl_map. (call np.asarray(..., dtype=) before calling the C function)

Reply:
the dtype of `refuniqshl_map` is already set to numpy.int32 in the function generating it `_get_refuniq_map`. See L171.

Do I need to add another `np.asarray(..., dtype=)` right before calling the C function in `intor_j2c`?